spicy/toolchain/include/compiler/detail/codegen/parser-builder.h

```diff
@@ -454,6 +454,8 @@ class ParserBuilder {
      */
     void finishLoopBody(const Expression& cookie, const Location& l);
 
+    void feature(const type::Unit& unit, std::string_view feature, std::function<void()> f);
```

Comment:
I like this API for putting code behind a feature flag. Would suggest to rename it though so that it conveys that `f` is building code behind that feature flag. `guardFeatureCode`? `addCodeForFeature`? `featureBlock`?

Also add Doxygen comment.